test/unit/codegen/codegen_c_visitor.cpp

```diff
@@ -222,3 +222,48 @@ SCENARIO("Check instance variable definition order", "[codegen][var_order]") {
         }
     }
 }
+
+SCENARIO("Check parameter constness with VERBATIM block",
+         "[codegen][verbatim_variable_constness]") {
+    GIVEN("A mod file containing parameter range variables that are updated in VERBATIM block") {
+        std::string const nmodl_text = R"(
+            NEURON {
+                SUFFIX IntervalFire
+                RANGE invl, burst_start
+            }
+            PARAMETER {
+                invl = 10 (ms) <1e-9,1e9>
+                burst_start = 0 (ms)
+            }
+            INITIAL {
+                LOCAL temp
+                : as invl is used in verbatim, it shouldn't be treated as const
+                VERBATIM
+                invl = 11
+                ENDVERBATIM
+                : burst_start time is read-only and hence can be const
+                temp = burst_start
+            }
+        )";
+
+        THEN("Variable used in VERBATIM shouldn't be marked as const") {
+            std::stringstream ss;
+
+            /// parse mod file & print mechanism structure
+            const auto& ast = NmodlDriver().parse_string(nmodl_text);
+            auto cvisitor = create_c_visitor(ast, nmodl_text, ss);
+            cvisitor->print_mechanism_range_var_structure();
```

Comment:
I think this is fine for now. Maybe if this file gets too repetitive then we could try something like
```c++
auto const reindented_generated_code = invoke_with_c_visitor([](auto& cvisitor) {
  cvisitor->print_mechanism_range_var_structure();
});
```
which would also handle the lifetime issues.